l3kernel/l3luatex.dtx

```diff
@@ -554,6 +616,61 @@ end
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}{ltx.utils.require}
+%   Loads a Lua module. This function loads the module just like the standard
+%   Lua global function |require|, with two key differences:
+%   \begin{enumerate}
+%     \item This function is guaranteed not to throw an error.
+%     \item This function does \emph{not} return the loaded module; instead, it
+%           returns |nil| on success and an error message on failure.
+%   \end{enumerate}
+%    \begin{macrocode}
+local function require(name)
+  if package_loaded[name] then
+    return
+  end
+
+  local failure_details = {}
+  for _, searcher in ipairs(package_searchers) do
+    local loader, data = searcher(name)
+    if type(loader) == 'function' then
+      package_loaded[name] = loader(name, data) or true
+      return
+    elseif type(loader) == 'string' then
+      failure_details[#failure_details + 1] = loader
+    end
+  end
+
+  return table_concat(failure_details, '\n')
+end
+
+ltx.utils.require = require
+%    \end{macrocode}
+% \end{macro}
+%
+% \begin{macro}{\@@_load_module_p:n}
+%   Check to see if we can load a module using |ltx.utils.require|. If we can
+%   load the module, then we load it immediately. Otherwise, we save the error
+%   message in |\g_@@_err_msg_str|.
+%    \begin{macrocode}
+local char_given   = command_id'char_given'
+local c_true_bool  = token_create(1, char_given)
+local c_false_bool = token_create(0, char_given)
+local c_str_cctab  = token_create('c_str_cctab').mode
+
+luacmd('@@_load_module_p:n', function()
+  local result = require(scan_string())
+  if result == nil then
+    set_macro(c_str_cctab, 'g_@@_err_msg_str', '', 'global')
```

Comment:
Why global?

Reply:
Wasn't sure if there was any hidden grouping somewhere. Changed to local